src/terminal/Process.cpp

```diff
@@ -427,15 +427,17 @@ vector<string> Process::loginShell()
 #else
     if (passwd const* pw = getpwuid(getuid()); pw != nullptr)
     {
-        #if defined(__APPLE__)
+#if defined(__APPLE__)
+        auto shell = string(pw->pw_shell);
+        auto index = shell.rfind('/');
         return {
-            "/usr/bin/login",
-            "-fp",
-            pw->pw_name,
+            "/bin/bash",
+            "-c",
+            fmt::format("exec -a -{} {}", shell.substr(index + 1, 5), pw->pw_shell)
```

Comment:
quickly afk. let me get a drink before accepting the existence of such a hack. 🍺